src/Tribe/Views/V2/Views/Traits/HTML_Cache.php

```diff
@@ -196,4 +200,152 @@ protected function should_enable_html_cache( $context ) {
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param array   $nonces   List of action and field name where the nonce is stored.
+		 * @param Context $context  Context from the current view.
+		 * @param View    $view     Which view instance we are currently rendering.
+		 */
+		return apply_filters( 'tribe_events_views_v2_get_view_nonce_json_properties', $nonces, $this->get_context(), $this );
+	}
+
+	/**
+	 * Does string replacement on the HTML cached based on the possible places we look for cached nonce values to inject
+	 * the correct string placeholder so we can remove it later.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param string   $html  HTML with the nonces to be replaced.
+	 *
+	 * @return string  HTML after replacement is complete.
+	 */
+	protected function extract_nonces_before_cache( $html ) {
+		$nonce_fields = $this->get_view_nonce_fields();
+		$nonce_attrs  = $this->get_view_nonce_attributes();
+		$nonce_props  = $this->get_view_nonce_json_properties();
+
+		foreach ( $nonce_fields as $action => $field ) {
+			$html = preg_replace( '!(<input[^>]+name="' . preg_quote( $field ) . '"[^>]+value=")[^"]*("[^>]*>)!', '\1%%NONCE:' . $action . '%%\2', $html );
+			$html = preg_replace( '!(<input[^>]+value=")[^"]*("[^>]+name="' . preg_quote( $field ) . '"[^>]*>)!', '\1%%NONCE:' . $action . '%%\2', $html );
+		}
+
+		foreach ( $nonce_attrs as $action => $attr ) {
+			$html = preg_replace( '!(' . preg_quote( $attr ) . '=")[^"]*(")!', '\1%%NONCE:' . $action . '%%\2', $html );
+		}
+
+		foreach ( $nonce_props as $action => $prop ) {
+			$html = preg_replace( '!("' . preg_quote( $prop ) . '":")[^"]*(")!', '\1%%NONCE:' . $action . '%%\2', $html );
+		}
+
+		return $html;
+	}
+
+	/**
+	 * Does string replacement on the HTML cached based on the possible places we look for cached nonce values.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param string   $html  HTML with the nonces to be replaced.
+	 *
+	 * @return string  HTML after replacement is complete.
+	 */
+	protected function inject_nonces_into_cached_html( $html ) {
+		$nonce_fields = $this->get_view_nonce_fields();
+		$nonce_attrs  = $this->get_view_nonce_attributes();
+		$nonce_props  = $this->get_view_nonce_json_properties();
+
+		$nonce_actions = array_merge( array_keys( $nonce_fields ), array_keys( $nonce_attrs ), array_keys( $nonce_props ) );
+		$nonce_actions = array_unique( $nonce_actions );
+
+		foreach ( $nonce_actions as $nonce_action ) {
+			$nonce = $this->maybe_generate_nonce( $nonce_action );
+			$html  = str_replace( "%%NONCE:{$nonce_action}%%", $nonce, $html );
+		}
+
+		return $html;
+	}
+
+	/**
+	 * Get a generated nonce required for HTML cache replacement based on an action provided.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param string $action  Which action will be used to generate the nonce.
+	 *
+	 * @return string  Nonce based on action passed.
+	 */
+	protected function maybe_generate_nonce( $action ) {
+		$generated_nonces = tribe_get_var( __METHOD__, [] );
```

Comment:
:no_entry_sign: Short array syntax is not allowed

<sub>hash: 0884b0df021bac64628a976a0e9b3a4d</sub>